spec/measure_repository_service_test_kit/library_group_spec.rb

```diff
@@ -88,4 +88,389 @@
+      requested name' do
+      library = FHIR::Library.new(name: 'INVALID_NAME')
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?name=#{library_name}"
+      ).to_return(status: 200, body: bundle.to_json)
+
+      result = run(test, url:, library_name:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the search request does not return a FHIR searchset bundle' do
+      library =  FHIR::Library.new(name: 'INVALID_NAME')
+      stub_request(
+        :get,
+        "#{url}/Library?name=#{library_name}"
+      ).to_return(status: 200, body: library.to_json)
+
+      result = run(test, url:, library_name:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server successfully searches and retrieves specified resource by its title' do
+    let(:test) { group.tests[6] }
+    let(:library_title) { 'library_title' }
+
+    it 'passes if the libraries in the returned FHIR searchset bundle match the requested title' do
+      library = FHIR::Library.new(title: library_title)
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?title=#{library_title}"
+      ).to_return(status: 200, body: bundle.to_json)
+
+      result = run(test, url:, library_title:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if search does not return 200' do
+      library =  FHIR::Library.new(title: library_title)
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?title=#{library_title}"
+      ).to_return(status: 400, body: bundle.to_json)
+
+      result = run(test, url:, library_title:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the libraries in the returned FHIR searchset bundle do not match the
+      requested title' do
+      library = FHIR::Library.new(title: 'INVALID_TITLE')
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?title=#{library_title}"
+      ).to_return(status: 200, body: bundle.to_json)
+
+      result = run(test, url:, library_title:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the search request does not return a FHIR searchset bundle' do
+      library =  FHIR::Library.new(title: 'INVALID_TITLE')
+      stub_request(
+        :get,
+        "#{url}/Library?title=#{library_title}"
+      ).to_return(status: 200, body: library.to_json)
+
+      result = run(test, url:, library_title:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server successfully searches and retrieves specified resource by its status' do
+    let(:test) { group.tests[7] }
+    let(:library_status) { 'library_status' }
```

Comment:
Also a nitpick, but since status is a `code` type, we should use `draft` in case we want to switch status to a radio button or something...

Reply:
So this would become `let(:library_status) { 'draft' }`?

Reply:
Yep! that should work :)